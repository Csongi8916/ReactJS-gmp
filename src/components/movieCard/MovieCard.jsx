import React from 'react';
import { Link } from 'react-router-dom';
import { getYear, separateGenres } from '../../utils/movieUtility';
import './MovieCard.scss';

//import pic from '../../assets/testimg.jpg'

export default function MovieCard({ id, title, year, genres, image }) {
  return (
    <div className="MovCard">
      <Link to={{
        pathname: `/movie/${title}`,
        state: { id: id }
      }}
        className="mov-link">
        <img src={image} width="300" height="410" />
        <div className="Row">
          <span className="MovCard_MovieTitle">{title}</span>
          <div className="MovCard_MovieInfo MovCard_MovieInfoBorder">{getYear(year)}</div>
        </div>
        <p className="MovCard_MovieInfo">{separateGenres(genres)}</p>
      </Link>
    </div>
  )
}
