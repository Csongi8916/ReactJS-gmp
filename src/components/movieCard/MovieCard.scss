$infoColor: darken(#fff, 35%);
$nrRedColor: #f65261;

.MovCard {
  margin: 30px;
  max-width: 300px;

  .Row {
    margin-top: 10px
  }

  .MovCard_MovieTitle {
    flex-grow: 1;
  }

  .mov-link, .mov-link:active, .mov-link:hover {
    color: $infoColor;
    text-decoration: none;
  }

  .MovCard_MovieInfo {
    color: $infoColor;
    font-size: 12px;
  }

  .MovCard_MovieInfoBorder {
    border: 2px solid $infoColor;
    border-radius: 5px;
    max-height: 15px;
    padding: 1px 10px;
  }
}