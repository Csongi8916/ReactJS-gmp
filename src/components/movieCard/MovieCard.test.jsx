import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import MovieCard from './MovieCard';

describe('Snapshpt tests', () => {
  test('Should render correctly MovieCard', () => {
    const tree = renderer.create(
      <MovieCard
        key={1}
        title={'Black Panther'}
        image={'https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg'}
        year={'2018-02-13'}
        genres={[
          "Action",
          "Adventure",
          "Fantasy",
          "Science Fiction"
        ]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});