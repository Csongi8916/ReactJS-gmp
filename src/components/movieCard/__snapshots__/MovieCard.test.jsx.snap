// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshpt tests Should render correctly MovieCard 1`] = `
<div
  className="MovCard"
>
  <img
    height="410"
    src="https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg"
    width="300"
  />
  <div
    className="Row"
  >
    <span
      className="MovCard_MovieTitle"
    >
      Black Panther
    </span>
    <div
      className="MovCard_MovieInfo MovCard_MovieInfoBorder"
    >
      2018
    </div>
  </div>
  <p
    className="MovCard_MovieInfo"
  >
    Action & Adventure & Fantasy & Science Fiction
  </p>
</div>
`;
